### PR TITLE
feat: add allIdentical

### LINKED ISSUE #578: allIdentical
Same as `RA.allEqual` but using `R.identical` as a comparator.

### PR DESCRIPTION
**Pull request by char0n**
Closes #578

### CODE REVIEW

File: test/allIdentical.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+describe('allIdentical', function() {
+  context('given all items are identical', function() {
+    specify('returns true', function() {
+      assert.isTrue(RA.allIdentical([4, 4, 4, 4]));
+    });
+  });
+
+  context('given all items are deeply equal', function() {
+    specify('should return true', function() {
```
Comment (guillaumearm) [inline test/allIdentical.js]:
```suggestion
    specify('should return false', function() {
```

File: test/allIdentical.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    });
+  });
+
+  context('given items have different type', function() {
+    specify('should return false', function() {
+      assert.isFalse(RA.allIdentical([1, 1, '1', 1]));
+    });
+  });
+
+  context('given items are reference to function', function() {
```
Comment (guillaumearm) [inline test/allIdentical.js]: you can add : 
```js
  context('given items are same functions without the same reference', function() {
    specify('should return false', function() {
      const getF = () => () => {};
      assert.isFalse(RA.allIdentical([getF(), getF(), getF()]));
    });
  });
```

File: src/allIdentical.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @sig [a] -> Boolean
+ * @param {Array} list The list of values
+ * @return {boolean}
+ * @see {@link https://ramdajs.com/docs/#identical|identical}
+ * @example
+ *
+ * RA.allIdentical([ 1, 2, 3, 4 ]); //=> false
+ * RA.allIdentical([ 1, 1, 1, 1 ]); //=> true
+ * RA.allIdentical([]); //=> true
+ * RA.allIdentical([ {}, {} ]); //=> false
```
Comment (guillaumearm) [inline src/allIdentical.js]:
```js
RA.allIdentical([ () => {}, () => {} ]); //=> false
```

File: src/allIdentical.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+const allIdentical = curry1(
```
Comment (guillaumearm) [inline src/allIdentical.js]: Maybe we can make a more optimized version using `R.reduce` and `R.reduced` ?

On the other hand, the implementation is less beautiful.

Comment (char0n) [reply src/allIdentical.js]: We could as  a performance optimization after it is in master.

File: src/allIdentical.js (hunk 3 above)
Comment (char0n) [reply src/allIdentical.js]: will add

File: test/allIdentical.js (hunk 2 above)
Comment (char0n) [reply test/allIdentical.js]: make sense will add

**Comment by char0n** [comment]
Everything addressed. Thank you for review

**Pull request merged**